### PR TITLE
IBX-6431: `IsVirtual` Search Criterion added

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | (https://issues.ibexa.co/browse/IBX-6431)
| Versions      | master
| Edition       | Commerce

`IsVirtual` Search Criterion added.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/search/criteria_reference/isvirtual_criterion.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## Arguments
+
+- (optional) `value` - bool representing whether to search for virtual (default `true`)
+or physical (`false`) products.
+
+## Limitations
+
+The `IsVirtual` Criterion is available in Solr or Elasticsearch engines.
+
```
Comment (ciastektk) [inline docs/search/criteria_reference/isvirtual_criterion.md:13]: Criterion is available in all supported search engines so there are no limitations. Based on other described criterions for products, Limitations section should be added when at least one of engine is not supported.

**Review by ciastektk — CHANGES_REQUESTED** [review]

File: docs/search/criteria_reference/isvirtual_criterion.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# IsVirtual Criterion
+
+The `IsVirtual` Search Criterion searches for virtual products.
```
Comment (ciastektk) [inline docs/search/criteria_reference/isvirtual_criterion.md:3]:
```suggestion
The `IsVirtual` Search Criterion searches for virtual or physical products. 
```

File: docs/search/criteria_reference/isvirtual_criterion.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# IsVirtual Criterion
+
+The `IsVirtual` Search Criterion searches for virtual or physical products. 
+
+## Arguments
+
+- (optional) `value` - bool representing whether to search for virtual (default `true`)
```
Comment (ciastektk) [inline docs/search/criteria_reference/isvirtual_criterion.md:7]: There is no "value" argument: https://github.com/ibexa/product-catalog/pull/1025/files#diff-05ba6990cb58e046d2e1b37f039baf002c696413d8841c7e1411c0f9b8b1bcc8R17

**Review by ciastektk — CHANGES_REQUESTED** [review]

**Review by ciastektk — APPROVED** [review]

**Review by juskora — APPROVED** [review]

File: docs/search/criteria_reference/product_search_criteria.md [hunk 4]
```diff
@@ -26,6 +26,7 @@ Search Criterion let you to filter product by specific attributes, for example:
 |[FloatAttributeRange](floatattributerange_criterion.md)|Value of product's float attribute|
 |[IntegerAttribute](integerattribute_criterion.md)|Value of product's integer attribute|
 |[IntegerAttributeRange](integerattributerange_criterion.md)|Value of product's integer attribute|
+|[IsVirtual](isvirtual_criterion.md)|Product type (virtual)|
```
Comment (adriendupuis) [inline docs/search/criteria_reference/product_search_criteria.md:29]:
```suggestion
|[IsVirtual](isvirtual_criterion.md)|[Product type](products.md#product-types) (virtual or physical)|
```

**Review by adriendupuis — APPROVED** [review]
Approved. Just an optional suggestion, as there is no such thing elsewhere in https://doc.ibexa.co/en/latest/search/criteria_reference/product_search_criteria/ it might not be the place to start cross-linking this way.

**Pull request merged**